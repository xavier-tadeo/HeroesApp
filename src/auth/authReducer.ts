import { types } from "../types/types";

// const state = {
//   name: "arlet",
//   logged: true
// }

type TAction = {
  type: string;
  payload: {
    name: string;
    email: string;
  };
};

export const authReducer = (state = {}, action: TAction) => {
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
      };

    case types.logout:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
