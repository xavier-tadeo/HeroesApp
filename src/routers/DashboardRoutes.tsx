import { Route, Routes } from "react-router-dom";
import { DcScreen } from "../components/DcScreen/DcScreen";
import { MarvelScreen } from "../components/MarvelScreen/MarvelScreen";
import { Navbar } from "../components/Navbar/Navbar";
import { Search } from "../components/Search/Search";

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="marvel" element={<MarvelScreen />} />
        <Route path="dc" element={<DcScreen />} />
        <Route path="search" element={<Search />} />
        <Route path="/" element={<MarvelScreen />} />
      </Routes>
    </>
  );
};
