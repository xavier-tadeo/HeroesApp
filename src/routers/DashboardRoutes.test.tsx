import { mount } from "enzyme";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../auth/authContext";
import { DashboardRoutes } from "./DashboardRoutes";

describe("Given a Dashboard component", () => {
  const contextValue = {
    user: {
      logged: true,
      name: "user",
    },
  };
  describe("When", () => {
    test("Then should correcty form to Marvel", () => {
      const view = render(
        <AuthContext.Provider value={contextValue}>
          <MemoryRouter initialEntries={["/"]}>
            <DashboardRoutes />
          </MemoryRouter>
        </AuthContext.Provider>
      );

      const nameUserText = screen.getByRole("link", { name: "Home" });
      const nameDch1 = screen.getByRole("heading", { name: "Marvel" });

      expect(nameDch1).toBeInTheDocument();

      expect(view).toMatchSnapshot();
      expect(nameUserText).toBeInTheDocument();
    });
    test("Then should correcty form to Dc", () => {
      render(
        <AuthContext.Provider value={contextValue}>
          <MemoryRouter initialEntries={["/dc"]}>
            <DashboardRoutes />
          </MemoryRouter>
        </AuthContext.Provider>
      );

      const nameDch1 = screen.getByRole("heading", { name: "DC" });

      expect(nameDch1).toBeInTheDocument();
    });
  });
});
