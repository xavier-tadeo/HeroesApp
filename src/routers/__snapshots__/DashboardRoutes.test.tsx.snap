// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Dashboard component When Then should correcty form to Marvel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Home
        </a>
        <div
          class="navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-item nav-link"
              href="/marvel"
            >
              Marvel
            </a>
            <a
              class="nav-item nav-link"
              href="/dc"
            >
              DC
            </a>
            <a
              class="nav-item nav-link"
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
        >
          <ul
            class="navbar-nav ml-auto"
          >
            <span
              class="nav-item anv-link text-info"
            >
              user
            </span>
            <button
              class="nav-item nav-link btn"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
      <div>
        <h1>
          Marvel
        </h1>
        <div
          class="row row-cols-1 row-cols-md-3 g-3"
        >
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Spider Man"
                    class="card-img"
                    src="/assets/marvel-spider.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Spider Man
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-spider"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Captain America"
                    class="card-img"
                    src="/assets/marvel-captain.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Captain America
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-captain"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Iron Man"
                    class="card-img"
                    src="/assets/marvel-iron.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Iron Man
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-iron"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Thor"
                    class="card-img"
                    src="/assets/marvel-thor.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Thor
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-thor"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Hulk"
                    class="card-img"
                    src="/assets/marvel-hulk.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Hulk
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hulk"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Wolverine"
                    class="card-img"
                    src="/assets/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Wolverine
                    </h5>
                    <p
                      class="crad-text"
                    >
                      James Howlett
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-wolverine"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Daredevil"
                    class="card-img"
                    src="/assets/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Daredevil
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-daredevil"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Hawkeye"
                    class="card-img"
                    src="/assets/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Hawkeye
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hawkeye"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Cyclops"
                    class="card-img"
                    src="/assets/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Cyclops
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-cyclops"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="row mo-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Silver Surfer"
                    class="card-img"
                    src="/assets/marvel-silver.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Silver Surfer
                    </h5>
                    <p
                      class="crad-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      class="card-text"
                    >
                       
                      <small
                        class="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-silver"
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Home
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link"
            href="/marvel"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link"
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item anv-link text-info"
          >
            user
          </span>
          <button
            class="nav-item nav-link btn"
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
    <div>
      <h1>
        Marvel
      </h1>
      <div
        class="row row-cols-1 row-cols-md-3 g-3"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Spider Man"
                  class="card-img"
                  src="/assets/marvel-spider.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Captain America"
                  class="card-img"
                  src="/assets/marvel-captain.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Iron Man"
                  class="card-img"
                  src="/assets/marvel-iron.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Thor"
                  class="card-img"
                  src="/assets/marvel-thor.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Hulk"
                  class="card-img"
                  src="/assets/marvel-hulk.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Wolverine"
                  class="card-img"
                  src="/assets/marvel-wolverine.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    class="crad-text"
                  >
                    James Howlett
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Daredevil"
                  class="card-img"
                  src="/assets/marvel-daredevil.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Hawkeye"
                  class="card-img"
                  src="/assets/marvel-hawkeye.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Cyclops"
                  class="card-img"
                  src="/assets/marvel-cyclops.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="row mo-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Silver Surfer"
                  class="card-img"
                  src="/assets/marvel-silver.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    class="crad-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    class="card-text"
                  >
                     
                    <small
                      class="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
